client/src/lib/result.js

```diff
@@ -40,3 +47,83 @@ export function paddedAttemptResults(result, numberOfAttempts) {
   const attemptResults = result.attempts.map((attempt) => attempt.result);
   return padSkipped(attemptResults, numberOfAttempts);
 }
+
+// Events sorted by best don't need forecast view.
+// Fewest moves is currently unsupported
+// Only final rounds are supported
+export function forecastViewEnabled(round) {
+  return round.format.sortBy != "best" &&
+    round.competitionEvent.event.id != "333fm" &&
+    round.advancementCondition === null;
+}
+
+export function resultProjectedAverage(result, format) {
+  if (isComplete(result.average)) {
```

Comment:
It can also be DNF, so we want to check specifically for skipped:

```suggestion
  if (!isSkipped(result.average)) {
```